### PR TITLE
Add a start-deamon command that only starts the daemon (instead of start)

### PR DESCRIPTION
**Pull request by nsams**
depends on #19 

start now also connects to the deamon

this allows multiple calls to start that start different scripts

start-daemon is started as background process in start (if not yet running)

Major (incompatible) changes:
- start is not blocking anymore (and doesn't show logs anymore)
- changing pm config file isn't possible anymore

### CODE REVIEW

File: src/commands/start-daemon.command.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        scripts,
+        processes,
+        shuttingDown: false,
+        server: undefined,
+    };
+
+    if (existsSync("./.pm.sock")) {
+        console.log(
+            "Could not start dev-pm server. A '.pm.sock' file already exists. \nThere are 2 possible reasons for this:\nA: Another dev-pm instance is already running. \nB: dev-pm crashed and left the file behind. In this case please remove the file manually.",
+        );
```
Comment (Fabian-Fynn) [inline src/commands/start-daemon.command.ts:38]: Please use the log style used throughout the project:

```
console.log(`${colors.bgYellow.bold.black(" dev-pm ")} Could not start dev-pm server. A '.pm.sock' file already exists. \nThere are 2 possible reasons for this:\nA: Another dev-pm instance is already running. \nB: dev-pm crashed and left the file behind. In this case please remove the file manually.`);
```

File: src/index.ts [hunk 2]
```diff
@@ -3,15 +3,16 @@
 import { Command } from "commander";
 
 import { logs, restart, shutdown, start, status } from "./commands";
+import { startDaemon } from "./commands/start-daemon.command";
 
 const program = new Command();
 
 program
-    .command("start [pmConfigFilePath]")
+    .command("start")
```
Comment (Fabian-Fynn) [inline src/index.ts:11]: How does the config file get loaded now?

File: src/commands/start-daemon.command.ts (hunk 1 above)
Comment (nsams) [reply src/commands/start-daemon.command.ts:38]: See https://github.com/vivid-planet/dev-process-manager/pull/21#discussion_r988612152 - this is a good example where it does not have any benefit

File: src/index.ts (hunk 2 above)
Comment (Fabian-Fynn) [reply src/index.ts:11]: I just remembered that there was a discussion about that. I think we decided to not hardcode it so the file could be named differently. Nonetheless I prefer it this way anyway.

**Review by Fabian-Fynn — APPROVED** [review]

**Pull request merged**